pkg/controller/common/metadata.go

```diff
@@ -24,7 +24,7 @@ const (
 	// GenerationKey represents the generation to which the resource was last reconciled
 	GenerationKey = MetadataNamespace + "/generation"
 
-	// MeshGenerationKey represents the generation of the service mesh to which the resource was last reconciled
+	// MeshGenerationKey represents the generation of the service mesh operator to which the resource was last reconciled. The operator will look at the operator version and generation to determine whether a control plane should be upgraded.
```

Comment:
I don't like this wording.  The original wording is correct, so maybe we just use:
```
// MeshGenerationKey represents the generation of the service mesh to which the resource was last reconciled.  This is uniquely identifies an installation, incorporating the operator version and the smcp resource generation.
```

Reply:
That makes sense. Clarifies that it's the operator. Confusion was is this the service mesh operator or service mesh control plane version. Thanks.

Reply:
Yeah, it's actually the version of the created resource, which would be defined by user settings (smcp generation), translation to values.yaml (operator version), and chart version (operator version).  Those three things determine the version of the resource that should be installed.